src/node/interfaces.cpp

```diff
@@ -398,6 +399,21 @@ class NodeImpl : public Node
     ArgsManager& args() { return *Assert(Assert(m_context)->args); }
     ChainstateManager& chainman() { return *Assert(m_context->chainman); }
     NodeContext* m_context{nullptr};
+    bool validateProxyAddress(const std::string ipAddress) override
```

Comment:
For consistency it would be best if the same rules apply to the `-proxy=` config and the text that is accepted via the GUI. For the former the validation is happening here:

https://github.com/bitcoin/bitcoin/blob/f1bcf3edc5027b501616670db33d8be1f2cb5a11/src/init.cpp#L1140-L1168

and here

https://github.com/bitcoin/bitcoin/blob/f1bcf3edc5027b501616670db33d8be1f2cb5a11/src/init.cpp#L1513-L1526

I am not sure how relevant the first check is for the proxy because the second check ends up calling `SplitHostPort()` as well. Maybe ignore the first, move the second to a new function and call this new function from the GUI and from `init.cpp`?

Reply:
Ok, let me do some testing on this. Thanks!

Reply:
I'm planning to refactor the `init.cpp` file after adding the persistence for the proxy setting, so the logic aligns with how the value is read and parsed from the `settings.json` file.

Reply:
> I am not sure how relevant the first check is for the proxy because the second check ends up calling SplitHostPort() as wel

I’ll restore the call to `SplitHostPort()` as advised offline. This is necessary because, without it, `LookupNumeric` would apply a default port if none is provided, making an address entered without a port (like `127`) appear valid. I discovered this while testing the persistence of values in `settings.json` (a separate PR, which is nearly finished and will be pushed ASAP). For example, if a user enters `127` in the UI, it’s stored in the file without a port. While adding a default port could work, it might not be transparent for the user.

Another point we could discuss, if necessary, in the "setting persistence" PR (coming soon today) is why an IP like 127 remains valid even when a port is specified. This is because, as discussed with @vasil during his review of https://github.com/bitcoin-core/gui/pull/836, the IP validation is left to the OS (thru CService. For example, `$ ping 127` results in `PING 127 (0.0.0.127) 56(84) bytes of data`, and this behavior aligns with how Qt and `bitcoind` currently handle IPs.